### PR TITLE
add keep_aspect_ratio flag to load_img

### PR DESCRIPTION
**Pull request by qdbp**
### Summary

Adds a `keep_aspect_ratio` kwarg to `ImageDataGenerator.flow_from_directory` to allow keeping images at their original aspect ratio, padding them to `target_size` with the value specified by the new `cval` kwarg.

### Related Issues

This resuscitates keras-team/keras#4987 by popular demand.

### PR Overview

- [y ] This PR requires new unit tests [y/n] (make sure tests are included)
- [ y] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [y ] This PR is backwards compatible [y/n]
- [n ] This PR changes the current API [y/n] (all API changes need to be approved by fchollet)

### CODE REVIEW

File: keras_preprocessing/image.py [hunk 1]
```diff
@@ -527,7 +532,21 @@ def load_img(path, grayscale=False, color_mode='rgb', target_size=None,
                         interpolation,
                         ", ".join(_PIL_INTERPOLATION_METHODS.keys())))
             resample = _PIL_INTERPOLATION_METHODS[interpolation]
-            img = img.resize(width_height_tuple, resample)
+            if not keep_aspect_ratio:
+                img = img.resize(width_height_tuple, resample)
+            else:
+                img.thumbnail(width_height_tuple, resample)
```
Comment (salvas87) [inline keras_preprocessing/image.py]: If size is smaller than width_height_tuple, then thumbnail will contain small image with edges on the right and at the bottom. I'm sure it is possible to resize an image to minimal height/width keeping aspect ratio.

**Comment by ldeluigi** [comment]
years go by

**Comment by fchollet** [comment]
Closing outdated PR. Note that the Keras Preprocessing repository is no longer in use, instead please just use the dataset utilities provided in `tf.keras.utils` ([see docs](https://keras.io/api/data_loading/)).

This particular feature is supported by `image_dataset_from_directory` as the `crop_to_aspect_ratio` argument.
